io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -289,18 +289,184 @@ impl Lvol {
+        if let Some(path) = self.path() {
+            std::fs::remove_file(path)?;
+        }
+        Ok(())
+    }
+
+    fn subpath(&self) -> std::path::PathBuf {
+        self.lvs()
+            .subpath()
+            .join("replica/")
+            .join(self.uuid())
+            .with_extension("json")
+    }
+}
+
+#[async_trait(?Send)]
+pub trait LogicalVolume {
+    /// returns the name of the Logical Volume
+    fn name(&self) -> String;
+
+    /// returns the UUID of the Logical Volume
+    fn uuid(&self) -> String;
+
+    /// return lvs for the Logical Volume
+    fn lvs(&self) -> Lvs;
+
+    /// returns the pool name of the Logical Volume
+    fn pool_name(&self) -> String;
+
+    /// returns the pool uuid of the Logical Volume
+    fn pool_uuid(&self) -> String;
+
+    /// returns a boolean indicating if the Logical Volume is thin provisioned
+    fn is_thin(&self) -> bool;
+
+    /// returns a boolean indicating if the Logical Volume is read-only
+    fn is_read_only(&self) -> bool;
+
+    /// return the size of the Logical Volume in bytes
+    fn size(&self) -> u64;
+
+    /// Returns Lvol disk space usage.
+    fn usage(&self) -> LvolSpaceUsage;
+}
+
+#[async_trait(?Send)]
+pub trait SpdkLvol: LogicalVolume + Share {
     /// returns the underlying bdev of the lvol
-    pub(crate) fn as_bdev(&self) -> UntypedBdev {
-        Bdev::checked_from_ptr(self.as_inner_ref().bdev).unwrap()
+    fn as_bdev(&self) -> UntypedBdev;
+    // wipe the first 8MB if unmap is not supported on failure the operation
+    // needs to be repeated
+    async fn wipe_super(&self) -> Result<(), Error>;
+
+    /// returns a boolean indicating if the lvol is a snapshot
+    fn is_snapshot(&self) -> bool;
+
+    /// get/read a property from this lvol from disk
+    async fn get(&self, prop: PropName) -> Result<PropValue, Error>;
+
+    /// destroy the lvol
+    async fn destroy(mut self) -> Result<String, Error>;
+
+    /// write the property prop on to the lvol but do not sync the metadata yet.
+    async fn set_no_sync(
+        self: Pin<&mut Self>,
+        prop: PropValue,
+    ) -> Result<(), Error>;
+
+    /// write the property prop on to the lvol which is stored on disk
+    async fn set(
+        mut self: Pin<&mut Self>,
+        prop: PropValue,
+    ) -> Result<(), Error>;
+
+    /// callback executed after synchronizing the lvols metadata
+    extern "C" fn blob_sync_cb(sender_ptr: *mut c_void, errno: i32);
+
+    /// write the property prop on to the lvol which is stored on disk
+    async fn sync_metadata(self: Pin<&mut Self>) -> Result<(), Error>;
+
+    /// Format snapshot name
+    /// base_name is the nexus or replica UUID
+    fn format_snapshot_name(base_name: &str, snapshot_time: u64) -> String;
+
+    /// Create a snapshot
+    async fn create_snapshot(&self, nvmf_req: &NvmfReq, snapshot_name: &str);
+
+    /// Create snapshot for local replica
+    async fn create_snapshot_local(
```

Comment:
This one will also be removed later when implementing snapshots.

Reply:
Currently not handled as part of this PR. As discussed, We will handle it when the Snapshot Trait will be added.